src/movies.js

```diff
@@ -1,15 +1,119 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(array) {
+  const allDirectors = array.map(function (movie) {
+    return movie.director;
+  });
+  return allDirectors;
+}
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(array) {
+  const spielberg = array.filter(function (movie) {
+    if (
+      movie.director === 'Steven Spielberg' &&
+      movie.genre.indexOf('Drama') !== -1
+    ) {
+      return true;
+    }
+  });
+  return spielberg.length;
+}
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+// To skip an object without an array filter only ones with numbers then average
+
+function ratesAverage(array) {
+    if (array.length === 0) {
+      return 0;
+    }
+  
+    let valuesOnly = array.map(function (movies) {
+      return movies.rate;
+    });
+    console.log(valuesOnly);
+  
+    function bouncer(arr) {
```

Comment:
I think you made your life harder here, a simple reduce method with an if statement inside for the edge cases would have been enough.
But it is nice that you experimented :)